actions/reservation.py

```diff
@@ -27,3 +37,87 @@ def get_actual_after_buffer(self) -> datetime.timedelta:
+        )
+
+        ical_event = Event()
+        # This should be unique such that if another iCal file is created
+        # for the same reservation, it will be the same as the previous one.
+        uid = f"varaamo.reservation.{self.reservation.pk}@{site_name}"
+        summary = self.get_ical_summary(language=language)
+        description = self.get_ical_description(language=language)
+
+        ical_event.add(name=EventProperty.UID, value=uid)
+        ical_event.add(name=EventProperty.DTSTAMP, value=local_datetime())
+        ical_event.add(name=EventProperty.DTSTART, value=self.reservation.begin.astimezone(DEFAULT_TIMEZONE))
+        ical_event.add(name=EventProperty.DTEND, value=self.reservation.end.astimezone(DEFAULT_TIMEZONE))
+
+        ical_event.add(name=EventProperty.SUMMARY, value=summary)
+        ical_event.add(name=EventProperty.DESCRIPTION, value=description, parameters={"FMTTYPE": "text/html"})
+        ical_event.add(name=EventProperty.LOCATION, value=self.get_location_string())
+
+        cal = Calendar()
+        cal.add(CalendarProperty.VERSION, "2.0")
+        cal.add(CalendarProperty.PRODID, "-//Helsinki City//NONSGML Varaamo//FI")
+
+        # Non-standard parameters, which are required for legacy compatibility.
+        cal["X-WR-RELCALID"] = uid
+        cal["X-WR-CALNAME"] = summary
+        cal["X-WR-CALDESC"] = description
+        cal["X-WR-TIMEZONE"] = settings.TIME_ZONE
+
+        cal.add_component(ical_event)
+        return cal.to_ical()
+
+    def get_ical_summary(self, *, language: Lang = "fi") -> str:
+        unit: Unit = self.reservation.reservation_unit.first().unit
+        unit_name = get_attr_by_language(unit, "name", language)
+        return _("Reservation for %(name)s") % {"name": unit_name}
+
+    def get_ical_description(self, *, language: Lang = "fi") -> str:
+        reservation_unit: ReservationUnit = self.reservation.reservation_unit.first()
+        unit: Unit = reservation_unit.unit
+        begin = self.reservation.begin.astimezone(DEFAULT_TIMEZONE)
+        end = self.reservation.end.astimezone(DEFAULT_TIMEZONE)
+
+        title = _("Booking details")
+        reservation_unit_name = get_attr_by_language(reservation_unit, "name", language)
+        unit_name = get_attr_by_language(unit, "name", language)
+        address = self.get_location_string()
+        start_date = begin.date().strftime("%d.%m.%Y")
+        start_time = begin.time().strftime("%H:%M")
+        end_date = end.date().strftime("%d.%m.%Y")
+        end_time = end.time().strftime("%H:%M")
+        time_delimiter = "klo" if language == "fi" else "kl." if language == "sv" else "at"
+        from_ = _("From")
+        to_ = _("To")
+        footer = _(
+            "Manage your booking at Varaamo. You can check the details of your booking and Varaamo's "
+            "terms of contract and cancellation on the 'My bookings' page."
+        )
+
+        return (
+            f"<html>"
```

Comment:
? `<!doctype html>`

Reply:
Not strictly necessary, since the HTML descriptions don't need to be proper HTML documents, but added it just in case.